src/common/useTranslations.tsx

```diff
@@ -0,0 +1,96 @@
+import { Fragment } from "react";
+import { useTemplateData } from "src/common/useTemplateData";
+
+/**
+ * A translation function translates text using the JSON file in i18n/locale/ for the current locale.
+ * @param text The text to translate
+ * @param replacements A dictionary of replacements to make in the translated text.
+ *   The key is the placeholder to replace, and the value is the replacement.
+ *   The keys are expected to be in braces in the original text.
+ *   For example, for text "Hello {NAME}", the key should be "NAME".
+ *   The replacement value may be a string or a JSX element.
+ * @returns The translated text. If any JSX is inserted, it returns an array of JSX elements.
```

Comment:
Having this function guarantee a string return value if no JSX replacements are given lets us use the same function where only a string value is allowed, such as element attribute values.

Reply:
I'm not sure I understand the use case for this type. In your example at least (https://github.com/yextconsulting/pages-starter-react-consulting/compare/jronkin/i18n..jronkin/translations) it seems like we always us the returned `translate` function without passing in replacements and just use the translations from `useTemplateData`. Is there a situation where we want to pass in overrides?